libs/sdk-core/src/persist/transactions.rs

```diff
@@ -345,6 +393,20 @@ fn filter_to_where_clause(
         }
     }
 
+    if let Some(filters) = metadata_filters {
+        filters.iter().for_each(
+            |PaymentMetadataFilter {
+                 search_path,
+                 search_value,
+             }| {
+                where_clause.push(format!(
+                    "metadata->'$.{}' = '{}'",
```

Comment:
Since we are formatting here a string using external input we should validate the input for security reasons.
Best would be to use a prepared statement and pass these values as parameters (at least the value).
We can validate the json path using https://docs.rs/serde_json_path/latest/serde_json_path/struct.JsonPath.html

Reply:
I agree using a prepared statement, because that is enough to prevent SQL injections, but is parsing the JSON path necessary? The worst that can happen if someone specifies a bad path would just be a bad filter that does not catch anything, which is not an issue on our side.

What do you think?

Reply:
I think If we use a prepared statement then we don't need to parse the json path.